### PR TITLE
Collect assigned virtual cores for PowerVS Instances

### PR DESCRIPTION
**Pull request by Kuldip-Nanda**
Also correcting the get_storage_types call as we are getting region specific info

### CODE REVIEW

File: app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb [hunk 1]
```diff
@@ -45,7 +45,7 @@ def instances
       # saving hardware information (CPU, Memory, etc.)
       ps_hw = persister.hardwares.build(
         :vm_or_template  => ps_vmi,
-        :cpu_total_cores => Float(instance["processors"]).ceil,
+        :cpu_total_cores => Float(instance['virtualCores']['assigned']),
```
Comment (jaywcarman) [inline app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb:48]: `instance['virtualCores']['assigned']` should already be an integer

https://cloud.ibm.com/apidocs/power-cloud#pcloud-pvminstances-get

Comment (Kuldip-Nanda) [reply app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb:48]: the column is original a Float Jay. So anyways it will be typecast automatically to Float. Unless we want to change that.

**Comment by jaywcarman** [comment]
restarting Travis

File: app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb [hunk 2]
```diff
@@ -45,7 +45,7 @@ def instances
       # saving hardware information (CPU, Memory, etc.)
       ps_hw = persister.hardwares.build(
         :vm_or_template  => ps_vmi,
-        :cpu_total_cores => Float(instance["processors"]).ceil,
+        :cpu_total_cores => Integer(instance['virtualCores']['assigned']),
```
Comment (jaywcarman) [inline app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb:48]: `instance['virtualCores']['assigned']` is already an Integer type so there's no need to pass it to `Integer()`:
```
irb(main):039:0> instance['virtualCores']['assigned'].kind_of? Integer
=> true
```

We can do this in another commit, but the same goes for the next line with `instance["memory"]`.

**Review by jaywcarman — APPROVED** [review]

**Review by agrare — APPROVED** [review]
